StRoot/StarGenerator/FILT/FcsDYBGFilter.cxx

```diff
@@ -0,0 +1,120 @@
+#include "FcsDYBGFilter.h"
+
+#include "StarGenerator/EVENT/StarGenParticle.h"
+#include "StarGenerator/EVENT/StarGenEvent.h"
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <cmath>
+#include <vector>
+#include <algorithm>
+
+static const float ZFCS    = 710.16 + 13.9 + 15.0;
+static const float XFCSMin = 16.69;
+static const float XFCSMax = 16.69 + 22.0 * 5.542;
+static const float YFCS    = 34.0/2.0 * 5.542;
+static const float FVCUT   = 0;//2.5; //cm
+static const float ETCUT   = 0.8;
+static const float DRCUT   = 40.0;
+static const float DYETCUT = 1.0;
+static const float DYMASSCUT= 2.5;
+
+
+static int ntot=0;
+static int ngood=0;
+
+using namespace std;
+//_______________________________________________________________
+FcsDYBGFilter::FcsDYBGFilter():StarFilterMaker("fcsDYBGFilter") {
+    cout<<"FCS DYBG filter is used!!!"<<endl;
+    cout<<"FCS DYBG filter swap particles="<<mSwap<<endl;
+}
+
+FcsDYBGFilter::FcsDYBGFilter(int dy, int check, int swap):StarFilterMaker("fcsDYBGFilter"), mDYmode(dy), mCheckmode(check), mSwap(swap){
+    cout<<"FCS DYBG filter is used!!!"<<endl;
+    cout<<"FCS DYBG filter DYMode="<<mDYmode<<endl;
+    cout<<"FCS DYBG filter Checkmode="<<mCheckmode<<endl;
+    cout<<"FCS DYBG filter swap particles="<<mSwap<<endl;
+}
+//_______________________________________________________________
+Int_t FcsDYBGFilter::Filter( StarGenEvent *mEvent){
+    ntot++;
+    // Get a reference to the current event 
+    StarGenEvent& event = *mEvent;
+    
+    //event.Print();
+    int np=event.GetNumberOfParticles();
+    if(np <= 0) {return kError;}
+    
+    // apply DYBG conditions for events
+    TIter Iterator = event.IterAll();
+    StarGenParticle *p = 0;
+    vector<StarGenParticle*> forwardParticles;
+    while( ( p = (StarGenParticle*)Iterator.Next() ) ){
+	int pid = abs(p->GetId());
+//	if(p->GetStatus()!=1 || pid==111) continue;
+        if(p->GetStatus()!=1) continue;
+	if(p->GetPz()<0.0) continue; // +z direction only
+	if(p->pt()<ETCUT) continue; 
+	//simple box cut around FCS  (vertex here is in [mm])
+	float x = fabs (p->GetVx()/10.0 + p->GetPx() / p->GetPz() * (ZFCS - p->GetVz()/10.0));
+	if(x<XFCSMin+FVCUT || x>XFCSMax-FVCUT) continue;
+	float y = fabs (p->GetVy()/10.0 + p->GetPy() / p->GetPz() * (ZFCS - p->GetVz()/10.0));
+	if(y>YFCS-FVCUT) continue;	
+	forwardParticles.push_back(p);	
+    }
+    unsigned int size=forwardParticles.size();
+    if(size<2) return StarGenEvent::kReject;
+    UInt_t res = 0x10;
+    int accept=0;
+    std::vector<int> swap(100);
```

Comment:
This allocates a vector with 100 zero values that don't seem to be used in any way.
```suggestion
    std::vector<int> swap;
```